File: android/app/src/main/kotlin/com/example/news_app/native/NativeChannelRegistry.kt
package com.nytimes.news_app.native

import android.content.Context
import com.nytimes.news_app.native.device.DeviceNativeHandler
import io.flutter.embedding.engine.FlutterEngine
import com.nytimes.news_app.native.display.DisplayNativeHandler

object NativeChannelRegistry {
    private val handlers = mutableListOf<BaseNativeHandler>()

    fun setup(flutterEngine: FlutterEngine, context: Context) {
        // Register display handler
        registerHandler(DisplayNativeHandler(flutterEngine, context))

        // Register device handler
        registerHandler(DeviceNativeHandler(flutterEngine, context))

        handlers.forEach { it.initialize() }
        println("✅ Android: Native Channel Registry initialized with ${handlers.size} handlers")
    }

    private fun registerHandler(handler: BaseNativeHandler) {
        handlers.add(handler)
        println("📱 Android: Registered handler: ${handler.channelName}")
    }

    fun onDestroy() {
        println("🔥 Android: Destroying Native Channel Registry")
        handlers.forEach { it.onDestroy() }
        handlers.clear()
    }
}

File: ios/Runner/Native/NativeChannelRegistry.swift
import Flutter
import UIKit

class NativeChannelRegistry {
    private static var handlers: [BaseNativeHandler] = []
    
    static func setup(flutterEngine: FlutterEngine, context: AppDelegate) {
        // Register display handler
        registerHandler(DisplayNativeHandler(flutterEngine: flutterEngine, context: context))
        
        // Register device handler
        registerHandler(DeviceNativeHandler(flutterEngine: flutterEngine, context: context))
        
        // Initialize all handlers
        handlers.forEach { $0.initialize() }
        
        print("✅ Native Channel Registry initialized with \(handlers.count) handlers")
    }
    
    private static func registerHandler(_ handler: BaseNativeHandler) {
        handlers.append(handler)
        print("📱 Registered handler: \(handler.channelName)")
    }
    
    static func onDestroy() {
        print("🔥 Destroying Native Channel Registry")
        handlers.forEach { $0.onDestroy() }
        handlers.removeAll()
    }
}
